samples/drivers/gecko_adc/src/bq24250.c

```diff
@@ -0,0 +1,330 @@
+		} else {
+			printf("\tCharger VBUS is missing!\n");
+		}
+		no_vbus = true;
+	} else {
+		no_vbus = false;
+		switch (charger_status) {
+		case 0:
+			printf("\tCharger is ready to charge!\n");
+			break;
+		case 1:
+			printf("\tCharger is currently charging!\n");
+			break;
+		case 2:
+			printf("\tCharger is done charging!\n");
+			break;
+		case 3:
+			printf("\tCharger is in Fault mode!\n");
+			break;
+		default:
+			break;
+		}
+
+		switch (charge_fault_status) {
+		case 0:
+			printf("\tCharger is in normal mode!\n");
+			break;
+		case 1:
+			printf("\tCharger is in Input OVP mode!\n");
+			break;
+		case 2:
+			printf("\tCharger is in Input UVLO mode!\n");
+			break;
+		case 3:
+			printf("\tCharger is in Sleep mode!\n");
+			break;
+		case 4:
+			printf("\tBattery is in Temperature Fault mode!\n");
+			break;
+		case 5:
+			printf("\tBattery is in OVP mode!\n");
+			break;
+		case 6:
+			printf("\tBattery is in Thermal Shutdown mode!\n");
+			break;
+		case 7:
+			printf("\tBattery is in Timer Fault!\n");
+			break;
+		case 8:
+			printf("\tBattery is in No Battery Connected ");
+			printf("mode - Try reseating the connector!\n");
+			break;
+		case 9:
+			printf("\tBattery is in ISET short mode!\n");
+			break;
+		case 10:
+			printf("\tBattery is in Input Fault and LDO low mode!\n");
+			break;
+		default:
+			break;
+		}
+	}
+	return ret;
+}
+
+static void bq_intr_callback(const struct device *port,
+		struct gpio_callback *cb, uint32_t pins)
+{
+	bq_battery_charge_int_isr_count++;
+	bq_battery_charge_int_state_change = true;
+}
+
+static void bq_thread(void *a, void *b, void *c)
+{
+	ARG_UNUSED(a);
+	ARG_UNUSED(b);
+	ARG_UNUSED(c);
+
+	k_msleep(4000);
+
+	printf("%s - thread started\n", __func__);
+	uint8_t current_state = BQ_INIT;
+	int ret;
+
+	memset(i2c_read_data, 0, 20);
+
+	while (1) {
+		if (restart_setup) {
+			printf("%s - restart_setup\n", __func__);
+			current_state = ENABLE_HARDWARE;
+			restart_setup = false;
+		}
+
+		switch (current_state) {
+		case BQ_INIT:
+
+			/* Setup and configure the interrupt for BQ_INT */
+
+			if (!gpiof_dev) {
+				printf("GPIOF driver error\n");
+				return;
+			}
+
+			/* Setup GPIO input, and triggers on falling edge. */
+			ret = gpio_pin_configure(gpiof_dev, BQ_CHARGE_INT,
+				GPIO_INPUT | GPIO_INT_EDGE_BOTH);
+			if (ret) {
+				printf("Error configuring GPIO_F %d!\n",
+						BQ_CHARGE_INT);
+			}
+
+			gpio_init_callback(&gpio_batt_chrg_cb,
+					bq_intr_callback, BIT(BQ_CHARGE_INT));
+
+			ret = gpio_add_callback(gpiof_dev, &gpio_batt_chrg_cb);
+			if (ret) {
+				printf("Cannot setup callback!\n");
+			}
+
+			ret = gpio_pin_interrupt_configure(gpiof_dev,
+				BQ_CHARGE_INT, GPIO_INPUT | GPIO_INT_EDGE_BOTH);
+
+			if (!device_is_ready(i2c1_dev)) {
+				printf("I2C: device is not ready\n");
+			}
+
+			k_msleep(1000);
+
+			bq_battery_charge_int_isr_count = 0;
+			bq_battery_charge_int_state_change = false;
+			memset(i2c_read_data, 0, 10);
```

Comment:
Is this call's intention to clear the buffer's first half and preserve the second half? If so, a more obvious implementation could be devised. If not, the hardcoded value of 10 should be replaced with` sizeof(i2c_read_data)`.

Reply:
Done